### PR TITLE
fix: android34 support

### PR DESCRIPTION
**Pull request by PoornimaApptentive**
## Summary
 - When the app targets Android 34, it crashes with the current/latest build.
 - Starting with Android 34, registering a dynamic broadcast receiver requires setting the receiver flags. The Apptentive MParticle kit relies on MParticle's Broadcast service providers to register the Apptentive kit process. A fix has been added for devices targeting Android 34 or higher using the RECEIVER_EXPORTED flag. Testing confirmed that it only works with RECEIVER_EXPORTED. Using RECEIVER_NOT_EXPORTED prevents receiving any broadcast messages from MParticle.

 ## Testing Plan
 - This was tested locally using our MParticle test app, MeerkatFacts.
 - Additional testing was performed by targeting the test app to various Android OS versions, including version 34

### CODE REVIEW

**Comment by einsteinx2** [comment]
Thanks, I'll get this looked at and we'll try and get it released as soon as possible.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt [hunk 1]
```diff
@@ -12,18 +16,27 @@ import com.mparticle.MParticle
...
-                    if (intent.action?.startsWith(MParticle.ServiceProviders.BROADCAST_ACTIVE) == true) {
-                        Apptentive.registerApptentiveActivityInfoCallback(callback)
-                    }
-                }
-            }, filter)
+            val filter = IntentFilter(MParticle.ServiceProviders.BROADCAST_ACTIVE + MParticle.ServiceProviders.APPTENTIVE)
+            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU) {
+                callback.getApptentiveActivityInfo()
+                    ?.registerReceiver(broadcastReceiver, filter, RECEIVER_EXPORTED)
+                Log.d("ApptentiveKitUtils", "registerApptentiveActivityContext: kit is active SDK 33+")
```
Comment (Mansi-mParticle) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt:34]:
```suggestion
                Logger.debug("ApptentiveKitUtils", "registerApptentiveActivityContext: kit is active SDK 33+")
```

can you please use here Logger.debug

File: src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt [hunk 2]
```diff
@@ -12,18 +16,27 @@ import com.mparticle.MParticle
...
+        val broadcastReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context, intent: Intent) {
+                if (intent.action?.startsWith(MParticle.ServiceProviders.BROADCAST_ACTIVE) == true) {
+                    Apptentive.registerApptentiveActivityInfoCallback(callback)
+                }
+            }
+        }
         if (MParticle.getInstance()?.isKitActive(MParticle.ServiceProviders.APPTENTIVE) == true) {
             Apptentive.registerApptentiveActivityInfoCallback(callback)
+            Log.d("ApptentiveKitUtils", "registerApptentiveActivityContext: kit is active")
```
Comment (Mansi-mParticle) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt:28]:
```suggestion
            Logger.debug("ApptentiveKitUtils", "registerApptentiveActivityContext: kit is active")
```

can you please use here Logger.debug

File: src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt (hunk 1 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt]: Updated.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt (hunk 2 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKitUtils.kt]: Updated.

**Review by Mansi-mParticle — APPROVED** [review]
Looks good to me

**Comment by PoornimaApptentive** [comment]
> Looks good to me

Thank you! Is there anything else I should do? The merging is still blocked.

**Comment by einsteinx2** [comment]
I've merged this in. We currently don't plan to do any releases until after the holidays out of an abundance of caution during this time period, so this should go live early January unless it gets escalated.

**Comment by einsteinx2** [comment]
@PoornimaApptentive We were actually able to get a release out yesterday, so this is now live.

**Pull request merged**